mobile/native/src/ln_dlc/mod.rs

```diff
@@ -525,6 +532,136 @@ pub fn get_usable_channel_details() -> Result<Vec<ChannelDetails>> {
+                                .inner
+                                .get_collateral_and_expiry_from(dlc_channel.channel_id)?;
+
+                            position::handler::update_position_after_dlc_creation(
+                                order,
+                                accept_collateral,
+                                expiry_timestamp,
+                            )?;
+
+                            tracing::info!("Successfully recovered position from order.");
+                        }
+                        Err(e) => {
+                            tracing::error!("Could not recover position from order as no filling order was found! Error: {e:#}");
+                        }
+                    }
+                } else if matches!(
+                    dlc_channel.state,
+                    SubChannelState::Offered(_)
+                        | SubChannelState::Accepted(_)
+                        | SubChannelState::Finalized(_)
+                ) {
+                    recover_dlc().await?;
+                } else {
+                    tracing::warn!("The DLC is in a state that can not be recovered.");
+                }
+            }
+        }
+        _ => {}
+    }
+
+    Ok(())
+}
+
+/// Sends a pending RecoverDlc background task notification to the UI, allowing the UI to show a
+/// dialog with a spinner that the DLC protocol is still in progress.
+/// Also triggers the `periodic_check` to process any actions that might have been created after the
+/// channel reestablishment.
+///
+/// fixme(holzeis): We currently use different events for show the recovery of a dlc and the waiting
+/// for an order execution in the happy case (without an restart in between). Those events and
+/// dialogs should be aligned.
+async fn recover_dlc() -> Result<()> {
+    tracing::warn!("It looks like the app was closed while the protocol was not finished yet.");
+    event::publish(&EventInternal::BackgroundNotification(
+        BackgroundTask::RecoverDlc(TaskStatus::Pending),
+    ));
+
+    // fixme(holzeis): We are manually calling periodic check here to speed up the processing of
+    // pending actions. Note, this might also not speed up the process, as the coordinator might
+    // have to resend a message to continue the protocol. This should be fixed in `rust-dlc` and any
+    // pending actions should be processed immediately once the channel is ready instead of
+    // periodically checking if a pending action needs to be sent. Note, pending actions can only
+    // get created on channel reestablishment.
+    tokio::time::sleep(Duration::from_secs(5)).await;
+    let node = NODE.try_get().context("failed to get ln dlc node")?;
+    if let Err(e) = node.inner.periodic_check().await {
+        tracing::error!("Failed to process periodic check! Error: {e:#}");
+    }
+
+    // Sending a failed event with a delay. This event will be ignored by the UI if the task has
+    // been successful in the meantime.
```

Comment:
why is the delay needed? it would be helpful to have that information.

Reply:
Moved that piece to the frontend, I think its cleaner now. Thanks for highlighting that short coming. 👍